core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -808,6 +810,7 @@ class DAGScheduler(
     // will be posted, which should always come after a corresponding SparkListenerStageSubmitted
     // event.
     stage.latestInfo = StageInfo.fromStage(stage, Some(partitionsToCompute.size))
+    outputCommitCoordinator.stageStart(stage.id)
```

Comment:
I wonder if it wouldn't be better to use a `SparkListener` to reduce coupling. Although that would potentially introduce race conditions in the code (since `LiveListenerBus` fires events on a separate thread).